### ISSUE TITLE
test multiple changes

### ISSUE THREAD

**Issue by shaunlebron** [parinfer/parinfer.js#180]
Problems:

- test annotation syntax only allows _one_ change
- syntax can be appended to allow _multiple_ changes, but not _overlapping_ changes

To allow both multiple and overlapping changes, I will support multiple `in` blocks.

<pre>
```in
(foo
 +++
bar)
+++
```

```in
(foo
  bar)
++
```

```out
(foo
  bar)
```
</pre>

This parses to the following data:

```js
{
  text: "()",
  options: {
    changes: [
      {lineNo: 0, x:1, oldText: '', newText: 'foo\nbar'},  // from first `in`
      {lineNo: 1, x:0, oldText: '', newText: '  '} // from second `in`
    ]
  }
}
```

Some details:

- "after" state of an `in` block must match the "before" state of the next `in` block
- each `in` block maps directly to a `change` object

**Comment by shaunlebron** [comment]
I forgot that multiple changes are already supported.  For example:

```
  (foo
++
    bar)
++
```

For the sake of naive diff parsing, I suppose I only allowed a single contiguous diff per line.  So we still need support for _overlapping_ changes.

I updated the documentation and allowed multiple `in` blocks for this purpose: https://github.com/shaunlebron/parinfer/commit/d4195d8f26b6c92d65bff2af5b9d391001c251fd
